packages/sdk-effects/storage/src/lib/storage.effects.ts

```diff
@@ -0,0 +1,56 @@
+import { BrandedString, TokenStoreObject } from '@forgerock/sdk-types';
+
+export interface StorageConfig {
+  tokenStore: TokenStoreObject | 'localStorage' | 'sessionStorage';
+  prefix: string;
+}
+
+interface StorageReturnValue {
+  get: () => Promise<string | null>;
+  set: (value: string) => Promise<void>;
+  remove: (value: string) => Promise<void>;
+}
+type StorageName<Name extends string> = BrandedString<string, Name>;
+
+export function makeStorageName<Name extends string>(name: Name) {
+  return name as unknown as BrandedString<string, Name>; // this is required to get the type branded.
+}
+
+export function createStorage<Name extends string>(
```

Comment:
so i opted to use a branded type, its not required but we could re-organize this api so its more composable.

how we do that is up for question, we would probably need to implement some curried functions for something like this, or we can make the customTokenStore a part of the `config` (optional property) so one argument is required for the entire effect.

```
export const effect = (config) => pipe(makeStorageName, createStorage)
```

where `effect` is the api to consume. 

we can also not use a branded type, its just a safer way to make primitives. so if its not desired we can omit this

Reply:
I guess I'm just missing the _why_ here. What value does this pattern bring to the project?